Client/src/Pages/Details/index.jsx

```diff
@@ -0,0 +1,156 @@
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell, { tableCellClasses } from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { StatusLabel } from "../../Components/Label/";
+
+/**
+ * Creates a styled TableCell component.
+ * @param {object} theme - The theme object.
+ * @returns {JSX.Element} The styled TableCell component.
+ */
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+    backgroundColor: theme.palette.action.hover,
+    color: theme.palette.common.black,
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+/**
+ * Creates a styled TableRow component.
+ * @param {object} theme - The theme object.
+ * @returns {JSX.Element} The styled TableRow component.
+ */
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  '&:nth-of-type(even)': {
+    backgroundColor: theme.palette.common.white,
+  },
+  '&:last-child td, &:last-child th': {
+    border: 0,
+  },
+}));
+
+/**
+ * Creates a data object for the table row.
+ * @param {JSX.Element} name - The status label JSX element.
+ * @param {string} date - The date and time string.
+ * @param {string} message - The message string.
+ * @returns {object} The data object containing name, date, and message.
+ */
+function createData(name, date, message) {
+  return { name, date, message };
+}
+
+/**
+ * Customized table component displaying incident history.
+ * @returns {JSX.Element} The JSX element representing the customized table.
+ */
+export default function CustomizedTables() {
+  const theme = useTheme();
+
+  const rows = [
+    createData(<StatusLabel status="Down" customStyles={{ backgroundColor: '#fff9f9', borderColor: '#ffcac6', color: '#344054' }} />, '2024-03-14 21:41:09', 'HTTP 350 - NOK'),
+    createData(<StatusLabel status="Down" customStyles={{ backgroundColor: '#fff9f9', borderColor: '#ffcac6', color: '#344054' }} />, '2024-03-14 21:41:09', 'timeout of 48000ms exceeded'),
+    createData(<StatusLabel status="Cannot resolve" customStyles={{ backgroundColor: '#f2f4f7', borderColor: '#d2d6de', color: '#344054' }} />, '2024-03-14 21:41:09', 'timeout of 48000ms exceeded'),
+    createData(<StatusLabel status="Cannot resolve" customStyles={{ backgroundColor: '#f2f4f7', borderColor: '#d2d6de', color: '#344054' }} />, '2024-03-14 21:41:09', 'timeout of 48000ms exceeded'),
+    createData(<StatusLabel status="Down" customStyles={{ backgroundColor: '#fff9f9', borderColor: '#ffcac6', color: '#344054' }} />, '2024-03-14 21:41:09', 'HTTP 350 - NOK'),
+  ];
+
```

Comment:
This component will need to take a `Monitor` object as a Prop, so it might be nice to refactor this to actually take a prop instead of hardcoding data.

Then we can pass it dummy data and have it ready to be hooked up.  Sample data of a Monitor:

```
	"montior": {
		"_id": "66863aa2437936b46e225d1f",
		"userId": "66863a8f437936b46e225d1a",
		"name": "Google",
		"description": "Google",
		"status": false,
		"type": "http",
		"url": "https://www.google.com/404",
		"isActive": true,
		"interval": 60000,
		"createdAt": "2024-07-04T06:01:06.527Z",
		"updatedAt": "2024-07-04T06:02:00.250Z",
		"__v": 0,
		"checks": [
			{
				"_id": "66863ad8437936b46e225d6d",
				"monitorId": "66863aa2437936b46e225d1f",
				"status": false,
				"responseTime": 144,
				"expiry": "2024-07-04T06:02:00.247Z",
				"statusCode": 404,
				"createdAt": "2024-07-04T06:02:00.248Z",
				"updatedAt": "2024-07-04T06:02:00.248Z",
				"__v": 0
			},
			{
				"_id": "66863b14437936b46e225d79",
				"monitorId": "66863aa2437936b46e225d1f",
				"status": false,
				"responseTime": 140,
				"expiry": "2024-07-04T06:03:00.167Z",
				"statusCode": 404,
				"createdAt": "2024-07-04T06:03:00.168Z",
				"updatedAt": "2024-07-04T06:03:00.168Z",
				"__v": 0
			}
		]
	}
```

You can use this sample data to set up the logic you need to populare the component

Reply:
Awesome! Thank you my guy I will definitely do and use that. Thanks again.

Reply:
no problem, if you need more sample data let your monitor run for a bit and then hit the get monitor endpoint